src/Driver/IBMDB2/Statement.php

```diff
@@ -120,16 +119,6 @@ private function bind($position, &$variable, int $parameterType, int $dataType):
      */
     public function execute($params = null): ResultInterface
     {
-        if ($params === null) {
-            ksort($this->bindParam);
```

Comment:
`db2_execute()` expects `$params` to contain parameters in the same order as they were bound via `db2_bind_param()`. Since parameters are appended to `$this->bindParam` alongside binding them to the statement, this order is always correct: https://github.com/doctrine/dbal/blob/b178d6019497543ca326e7f10e84fde112057126/src/Driver/IBMDB2/Statement.php#L109-L116

Sorting parameters by key breaks the expected order if the parameters were bound out of order.